### PR TITLE
update for newer llvm syntax

### PR DESCRIPTION
**Pull request by glguida**

### CODE REVIEW

**Comment by LdBeth** [comment]
Ah, sorry for the late response. Feel free to merge the changes I made.

The official ML1 website now has binary for x86_68 mach-o (lacking 64 bit support used to be my reason for using this LLVM method), but it is still interested to use a copy of ML1 that is “assembled” rather than compiled from C.

File: ml1.c [hunk 1]
```diff
@@ -214,10 +214,18 @@ mdread(uint8_t *c)
 	int inno = SVAR(10);
 	if ( inno == 0 )
 		return 1;
-	if ( inno > infs ) {
+	if ( inno > infs + 100 ||
+		(inno <= 100 && inno > infs) ||
```
Comment (glguida) [inline ml1.c:222]: Where does 100 comes from?

**Review by glguida — COMMENTED** [review]
Just a comment, this is great. I've been out of the loop on ML/I for a while so I just would like to understand.

Re SE23, any link to the spec for it?

**Comment by LdBeth** [comment]
The subtract 100 rule is from the Unix version manual (Appendix R) https://www.ml1.org.uk/htmldoc/ml1appr.html
and it is implemented on all C-map based versions.

I was looking for a way to include external files, and found this can be sort of a work around so I don't need to use `m4` for another pass, by just `cat` the external files into a single include file and use the rewind feature as a work around.

And I have download the macOS version from ml1.org.uk and compared to confirm that is the same behavior.

The newer C-map versions also has a new character variable feature not present in the LOWL AJB version, as I imaging future updates are exclusively made to the mapped C files, not the LOWL files. Right now I have not understood the LOWL source good enough to see if update to LOWL AJB can be done.

The listing feature I think is of little use, the input translation `S16` `S17` is also rather limited.

This version still works great so far on LLVM 17, and can be cross compiled (as I don't have complete LLVM installation on my Intel macbook, I use another Linux computer to convert llvm source to assembly.) The performance is a bit faster than the C version, and for some uninvestigated reason with same workspace size this translated version can hold longer text argument than the C version.

**Comment by glguida** [comment]
Glad you find this useful. I think it was written over a week more than 10 years ago. I really needed a powerful macro language, and found ML/I to be incredibly useful.

I also noticed that it is faster than the C version. I would guess LLVM is good at its job.

I think would be cool to be able to list this as an official port on ml1.org.uk, but never got to actually do that. I also think that there could be some license issues on directly adding the AJB sources here, which is why I never did it.

All things that should be reconsidered.

Thank you for your interest on this and for pushing the patches!

**Pull request merged**